server/src/main/java/org/opensearch/common/settings/ClusterSettings.java

```diff
@@ -162,6 +162,10 @@ public ClusterSettings(final Settings nodeSettings, final Set<Setting<?>> settin
         addSettingsUpdater(new LoggingSettingUpdater(nodeSettings));
     }
 
+    public void registerSetting(Setting<?> setting) {
+        super.registerSetting(setting);
```

Comment:
Do you need to `validateSettingKey(setting);` here? You do it elsewhere.

Reply:
Great question. 
IndexScopedSettings have specific validations which are done. But for ClusterSettings there are no specific validations, it just relies on AbstractScopedSettings. 

That said, I missed validating the settings key in AbstractScopedSettings. I'll go add that which validates ClusterSettings as well.